src/main/java/com/eerussianguy/blazemap/feature/maps/WorldMapGui.java

```diff
@@ -215,6 +231,33 @@ public boolean keyPressed(int key, int scancode, int modifiers) {
             return true;
         }
 
+        if (key == BlazeMapFeaturesClient.KEY_WAYPOINTS.getKey().getValue()) {
+            Level level = Minecraft.getInstance().level;
+            int posY = level == null ? 60 : level.getSeaLevel();
+            Coordination coordination = map.getCoordination();
+            BlockPos position = new BlockPos(coordination.blockX, posY, coordination.blockZ);
+            ChunkPos chunkPos = new ChunkPos(position);
+
+            // Attempt to get y actual level from MDCache
+            ChunkMDCache mdCache = ClientEngine.getMDCache(chunkPos);
+            if (mdCache != null) {
+                ChunkMDCacheView mdView = new ChunkMDCacheView().setSource(mdCache);
+                Set<BlazeRegistry.Key<DataType>> filterKeys = new HashSet<>();
+                filterKeys.add(UnsafeGenerics.stripKey(BlazeMapReferences.MasterData.TERRAIN_HEIGHT));
+                mdView.setFilter(filterKeys);
+
+                TerrainHeightMD heightMD = (TerrainHeightMD) mdView.get(BlazeMapReferences.MasterData.TERRAIN_HEIGHT);
```

Comment:
You don't need the view here. You can get the MD directly from the `ChunkMDCache`